providers/pear_channel.rb

```diff
@@ -24,25 +24,29 @@
 require 'chef/mixin/language'
 include Chef::Mixin::ShellOut
 
+use_inline_resources
+
 def whyrun_supported?
   true
 end
 
 action :discover do
   unless exists?
     Chef::Log.info("Discovering pear channel #{@new_resource}")
-    execute "#{node['php']['pear']} channel-discover #{@new_resource.channel_name}" do
+    r = execute "#{node['php']['pear']} channel-discover #{@new_resource.channel_name}" do
```

Comment:
You shouldn't need to do any of the updated_by_last action mess if you use inline resources.  The inline resource method handles bubbling up the notifications